.semaphore/semaphore.yml

```diff
@@ -487,6 +487,127 @@ blocks:
       commands:
       - make ci
 
+- name: "cni-plugin: Build Windows binaries"
+  run:
+    when: "branch =~ '.*' or change_in(['/*', '/cni-plugin/', '/libcalico-go/'], {exclude: ['/**/.gitignore', '/**/README.md', '/**/LICENSE']})"
+  dependencies: []
+  task:
+    prologue:
+      commands:
+      - cd cni-plugin
+    jobs:
+    - name: "build Windows binaries"
+      commands:
+      - make build && make bin/windows/win-fv.exe
+
+- name: "cni-plugin: Windows FV - docker"
+  run:
+    when: "branch =~ '.*' or change_in(['/*', '/cni-plugin/', '/libcalico-go/'], {exclude: ['/**/.gitignore', '/**/README.md', '/**/LICENSE']})"
+  dependencies: ["cni-plugin: Build Windows binaries"]
+  task:
+    secrets:
+      - name: banzai-secrets
+      - name: private-repo
+    prologue:
+      commands:
+        # Load the github access secrets.  First fix the permissions.
+        - chmod 0600 ~/.keys/*
+        - ssh-add ~/.keys/*
+        # Prepare aws configuration.
+        - pip install --upgrade --user awscli
+        - export REPORT_DIR=~/report
+        - export LOGS_DIR=~/fv.log
+        - export SHORT_WORKFLOW_ID=$(echo ${SEMAPHORE_WORKFLOW_ID} | sha256sum | cut -c -8)
+        - export CLUSTER_NAME=sem-${SEMAPHORE_PROJECT_NAME}-pr${SEMAPHORE_GIT_PR_NUMBER}-${SHORT_WORKFLOW_ID}-docker
+        - export KEYPAIR_NAME=${CLUSTER_NAME}
+        - echo CLUSTER_NAME=${CLUSTER_NAME}
+        - sudo apt-get install putty-tools
+        - git clone git@github.com:tigera/process.git ~/process
+        - cd cni-plugin
+        - make build && make bin/windows/win-fv.exe
+    epilogue:
+      always:
+        commands:
+          - artifact push job ${REPORT_DIR} --destination semaphore/test-results --expire-in ${SEMAPHORE_ARTIFACT_EXPIRY} || true
+          - artifact push job ${LOGS_DIR} --destination semaphore/logs --expire-in ${SEMAPHORE_ARTIFACT_EXPIRY} || true
+          - aws ec2 delete-key-pair --key-name ${KEYPAIR_NAME} || true
+          - cd ~/process/testing/winfv && NAME_PREFIX="${CLUSTER_NAME}" ./setup-fv.sh -q -u
+    env_vars:
+      - name: SEMAPHORE_ARTIFACT_EXPIRY
+        value: 2w
+      - name: AWS_DEFAULT_REGION
+        value: us-west-2
+      - name: MASTER_CONNECT_KEY_PUB
+        value: master_ssh_key.pub
+      - name: MASTER_CONNECT_KEY
+        value: master_ssh_key
+      - name: WIN_PPK_KEY
+        value: win_ppk_key
+      - name: K8S_VERSION
+        value: 1.17.2
+    jobs:
+      - name: FV Test matrix
+        execution_time_limit:
+          minutes: 60
+        commands:
+          - ./.semaphore/run-win-fv.sh
+
+- name: "cni-plugin: Windows FV - containerd"
+  run:
+    when: "branch =~ '.*' or change_in(['/*', '/cni-plugin/', '/libcalico-go/'], {exclude: ['/**/.gitignore', '/**/README.md', '/**/LICENSE']})"
+  dependencies: ["cni-plugin: Build Windows binaries"]
+  task:
+    secrets:
+      - name: banzai-secrets
+      - name: private-repo
```

Comment:
As we've discussed, and not for this PR, but in the near future we need to remove the dependency on a private repo for these.